enhancements/monitoring/integration-with-topology-modes.md

```diff
@@ -0,0 +1,176 @@
+---
+title: monitoring-topology-mode
+authors:
+  - "@lilic"
+  - "@simonpasquier"
+reviewers:
+  - "@openshift/openshift-team-monitoring"
+  - "@dhellmann"
+  - "@romfreiman"
+  - "@wking"
+approvers:
+  - "@openshift/openshift-team-monitoring"
+creation-date: 2021-03-01
+last-updated: 2021-03-01
+status: implementable
+see-also:
+  - "/enhancements/cluster-high-availability-mode-api.md"
+---
+
+# Monitoring Stack with Single Replica Topology Mode
+
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Cluster Monitoring Operator (CMO) needs to respect the
+`infrastructureTopology` field newly added to the
+`infrastructures.config.openshift.io` CRD. Depending on the value, it should
+configure the monitoring operands for highly-available operations or not.
+
+## Motivation
+
+As described in the [Cluster High-availability Mode API](https://github.com/openshift/enhancements/blob/master/enhancements/single-node/cluster-high-availability-mode-api.md)
+enhancement and implemened in [OpenShift API](https://github.com/openshift/api/pull/827),
+OpenShift 4.8 introduces 2 new status fields in the Infrastructure API that
+informs operators about the cluster deployment topology.
+
+Depending on whether an operand is deployed on the control nodes or not, the
+operator should look at either `controlPlaneTopology` and
+`infrastructureTopology`. If the field value is `HighlyAvailable`, the operator
+should configure the operand for high-avaible operation. If the field value is
+`SingleReplica`, it should *not* configure the operand for high-avaible
+operation.
+
+### Goals
+
+- Deploy the monitoring components with one replica for non-HA configuration.
+- Deploy the user-workload components with one replica for non-HA configuration.
+- Ensure that CMO passes the single-node tests from openshift/origin.
+
+### Non-Goals
+
+- Implement additional enhancements to reduce CPU and memory footprints.
+- Allow cluster admins to control the number of replicas per operand.
+- Address CodeReady Containers requirements (though it might help).
+- Add tests for `SingleReplica` to the existing end-to-end test suite in the CMO repository.
+
+## Proposal
+
+Whenever CMO needs to reconcile, it retrieves the
```

Comment:
does this mean it won't watch the topology resouce directly for changes?  It seems like it should, so we aren't dependent on a reconcile cycle triggered by something else to pick up a change.

Reply:
good point!